provider/aws/aws.go

```diff
@@ -75,19 +84,58 @@ func (p AWSProvider) String() string {
 	return ProviderName
 }
 
-func (p *AWSProvider) Ensure(configs map[provider.Resource]map[string]struct{}) error {
-	params := &cloudformation.DescribeStacksInput{
-		StackName: aws.String(stackName),
+func generateRoutes(configs map[provider.Resource]map[string]*net.IPNet) []string {
+	cidrs := make([]*net.IPNet, 0, len(configs))
+	for _, rs := range configs {
+		for _, ipnet := range rs {
+			cidrs = append(cidrs, ipnet)
+		}
 	}
-	resp, err := p.cloudformation.DescribeStacks(params)
-	if err != nil {
-		if aerr, ok := err.(awserr.Error); !ok || aerr.Code() == cloudformation.ErrCodeStackInstanceNotFoundException {
-			return err
+
+	sort.Slice(cidrs, func(i, j int) bool {
+		countI := cidr.AddressCount(cidrs[i])
+		countJ := cidr.AddressCount(cidrs[j])
+		if countI == countJ {
+			return cidrs[i].String() < cidrs[j].String()
 		}
+		return countI < countJ
+	})
+
+	newCIDRs := make([]string, 0, len(cidrs))
+	i := 0
+	for _, c := range cidrs {
+		contained := false
+		if i < len(cidrs)-1 {
+			for _, block := range cidrs[i+1:] {
+				if networkContained(c, block) {
+					contained = true
+					break
+				}
+			}
+		}
```

Comment:
I am not super happy with that, because CIDR has nothing to do with AWS and it could be generic and the logic can b reused in other providers, too.
What do you think?

Reply:
Sure, I can move it to another more generic place. Any preference where to put such a function? I could put it in the `provider` package, but maybe you prefer if I create a dedicated package for it?

Reply:
Lets start with provider package.

Reply:
Moved to provider package in `provider/cidr.go`